asserts/ifacedecls.go

```diff
@@ -342,19 +343,48 @@ func baseCompileRule(context string, rule interface{}, target rule, subrules []s
 	// compile and set subrules
 	for _, subrule := range subrules {
 		v := rMap[subrule]
-		if v == nil {
+		var l []interface{}
+		alternatives := false
+		switch x := v.(type) {
+		case nil:
 			v = defaultOutcome[subrule]
 			defaultUsed++
+		case []interface{}:
+			alternatives = true
+			l = x
+		}
+		if l == nil {
+			l = []interface{}{v}
```

Comment:
Shouldn't this be inside the nil case above?  Is there a situation where we can get `l == nil` as a desired consequence of `x == nil`?